src/wmtk/quadrature/ClippedQuadrature.cpp

```diff
@@ -0,0 +1,77 @@
+#include "ClippedQuadrature.h"
+
+#include <wmtk/utils/PolygonClipping.h>
+#include "TriangleQuadrature.h"
+
+namespace wmtk {
+
+void ClippedQuadrature::clipped_triangle_box_quadrature(
+    const int order,
+    const TriangleVertices& triangle,
+    const Eigen::AlignedBox2d& box,
+    Quadrature& quadr,
+    Quadrature* tmp)
+{
+#if 0
+    // Naive implementation for testing
+    PolygonVertices poly(4, 2);
+    poly.row(0) = box.corner(Eigen::AlignedBox2d::CornerType::BottomLeft).transpose();
+    poly.row(1) = box.corner(Eigen::AlignedBox2d::CornerType::BottomRight).transpose();
+    poly.row(2) = box.corner(Eigen::AlignedBox2d::CornerType::TopRight).transpose();
+    poly.row(3) = box.corner(Eigen::AlignedBox2d::CornerType::TopLeft).transpose();
+    clipped_triangle_polygon_quadrature(order, triangle, poly, quadr);
+#else
+    auto clipped = clip_triangle_by_box(triangle, box);
+
+    Quadrature dummy;
+    if (tmp == nullptr) {
+        tmp = &dummy;
+    }
+
+    TriangleQuadrature rules;
+    quadr.set_dimension(2);
+    quadr.clear();
+    for (int i = 1; i + 1 < clipped.rows(); ++i) {
```

Comment:
This is doing a triangle fan around v0. For a polygon with N vertices, we end up with N-2 triangles. I prefer to use `i + 1` here since it's safer when the size is an unsigned int and we have an empty polygon (not the case here, but it's just a habit). Compiler will optimize the repeated +1 anyway.